src/servers/src/grpc/flight/stream.rs

```diff
@@ -89,6 +89,15 @@ impl FlightRecordBatchStream {
                         warn!(e; "stop sending Flight data");
                         return;
                     }
+                    if let Some(metrics) = recordbatches
+                        .metrics()
+                        .and_then(|m| serde_json::to_string(&m).ok())
```

Comment:
maybe send metrics every X rows or every X recordbatch?

Reply:
Updating metrics too frequently may affect performance.

Reply:
The poll is based on record batch, only send right after every record batch can ensure the metrics must exists. The overhead is a consideration. I'm also hesitate for this.